### ISSUE TITLE
Search returns too many results

### ISSUE THREAD

**Issue by Popolechien** [openzim/sotoki#261]
Currently searching for string `toto` returns all results pages including `toto`, including comments and answers. 

Considering that the snippets of text shown are too short / not always very clear, it might be helpful to limit the results page to only return **questions** that include the target string.

**Comment by rgaudin** [comment]
I disagree with your diagnostic and solution but agrees on the root cause: search is a bit of a pain to use in its current form while being the recommended way to access posts.

[SO's advanced search feature](https://stackoverflow.com/search?q=NotImplementedError) is great and ideally, we'd need something similar:

<img width="759" alt="Screen Shot 2022-06-22 at 13 40 59" src="https://user-images.githubusercontent.com/57929/175043493-45b32240-8a75-44ad-bf44-584e48bc3d4b.png">

Search results helps figure out which result to click on:

<img width="751" alt="Screen Shot 2022-06-22 at 13 41 34" src="https://user-images.githubusercontent.com/57929/175043618-383fd236-e4af-4646-a069-3589a8ffb370.png">

- searches on Question title (that's a lot different than searching on whole question). Kiwix suggestions only work on exactly matching string LTR so it's useless in this case.
- shows date which is highly valuable
- shows language tags helping identifying if relevant or not
- shows score and number of answers informing about potential outcome
- can be sorted by score or date

Advanced-search-like feature is unlikely to happen in the foreseeable future but returning result snippets with this information would be a huge improvement… that's not possible ATM as we only have control over *what-to-index* (we don't currently but should soon)… but that might be a path to improvement in Kiwix.
